### PR TITLE
Cross-platform VPC network query API

### PR DESCRIPTION
**Pull request by siliconbrain**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | none
| License         | Apache 2.0

### What's in this PR?
This PR implements a cross-platform, VPC network related resource (network, subnetwork, route table) querying API, whose primary goal is to enable UIs to present options for the user when selecting existing resources for cluster creation. Currently Google and Amazon is supported as provider, but more are coming.

### Why?
There was no such cross-platform API available.

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [x] OpenAPI and Postman files updated, client regenerated (`make generate-client`) (if needed)
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: api/network.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+	"github.com/banzaicloud/pipeline/auth"
+	"github.com/banzaicloud/pipeline/internal/platform/gin/correlationid"
+	ginutils "github.com/banzaicloud/pipeline/internal/platform/gin/utils"
+	"github.com/banzaicloud/pipeline/internal/providers"
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+)
+
+// NetworkAPI implements network functions
+type NetworkAPI struct{}
```
Comment (sagikazarmark) [inline api/network.go]: Would be nice to inject dependencies here instead of accessing the global versions. Those should eventually go away.

Comment (siliconbrain) [reply api/network.go]: Which dependencies do you mean? Could you give an example? I only found the secret store as a global dependency, but that has a "private" type, so it's not trivial to create a reference to it.

Comment (sagikazarmark) [reply api/network.go]: I mean the logger, errorHandler, db connection dependencies.

Comment (siliconbrain) [reply api/network.go]: I added the logger, but couldn't find any more dependencies.

**Pull request merged**